skia.lua

```diff
@@ -0,0 +1,1021 @@
+    "src/shaders/gradients/SkRadialGradient.cpp",
+    "src/shaders/gradients/SkSweepGradient.cpp",
+    "src/shaders/gradients/SkTwoPointConicalGradient.cpp",
+    "src/shaders/SkBitmapProcShader.cpp",
+    "src/shaders/SkColorFilterShader.cpp",
+    "src/shaders/SkColorShader.cpp",
+    "src/shaders/SkComposeShader.cpp",
+    "src/shaders/SkImageShader.cpp",
+    "src/shaders/SkLightingShader.cpp",
+    "src/shaders/SkLights.cpp",
+    "src/shaders/SkLocalMatrixShader.cpp",
+    "src/shaders/SkPerlinNoiseShader.cpp",
+    "src/shaders/SkPictureShader.cpp",
+    "src/shaders/SkShader.cpp",
+    "src/utils/SkDashPath.cpp",
+    "src/utils/SkEventTracer.cpp",
+    "src/utils/SkMatrix22.cpp",
+    "src/utils/SkPatchUtils.cpp",
+    "src/utils/SkPolyUtils.cpp",
+    "src/utils/SkShadowTessellator.cpp",
+    "src/utils/SkShadowUtils.cpp",
+    "src/utils/SkUTF.cpp",
+    "third_party/gif/SkGifImageReader.cpp",
+    "third_party/skcms/skcms.cc",
+  }
+
+  -- Add common Win and WinUWP
+
+  local common_win = {
+    "src/core/SkSharedMutex.cpp",
+    "src/fonts/SkFontMgr_indirect.cpp",
+    "src/ports/SkDebug_win.cpp",
+    "src/ports/SkFontMgr_win_dw.cpp",
+    "src/ports/SkFontMgr_win_dw_factory.cpp",
+    "src/ports/SkOSFile_win.cpp",
+    "src/ports/SkScalerContext_win_dw.cpp",
+    "src/ports/SkTLS_win.cpp",
+    "src/ports/SkTypeface_win_dw.cpp",
+    "src/utils/win/SkDWrite.cpp",
+    "src/utils/win/SkDWriteFontFileStream.cpp",
+    "src/utils/win/SkDWriteGeometrySink.cpp",
+    "src/utils/win/SkHRESULT.cpp",
+  }
+
+  -- Add common Unix
+
+  local common_unix = {
+    "src/ports/SkDebug_stdio.cpp",
+    "src/ports/SkOSFile_posix.cpp",
+    "src/ports/SkTLS_pthread.cpp",
+  }
+
+  -- Add common Linux and Android
+
+  local common_linux = {
+    common_unix,
+    "src/core/SkTSearch.cpp",
+    "src/ports/SkFontMgr_fontconfig.cpp",
+    "src/ports/SkFontMgr_fontconfig_factory.cpp",
+    "src/ports/SkFontConfigInterface_direct.cpp",
+    "src/ports/SkFontHost_FreeType.cpp",
+    "src/ports/SkFontHost_FreeType_common.cpp",
+  }
+
+  local common_android = {
+    common_unix,
+    "src/core/SkTSearch.cpp",
+    "src/ports/SkDebug_android.cpp",
+    "src/ports/SkFontHost_FreeType.cpp",
+    "src/ports/SkFontHost_FreeType_common.cpp",
+    "src/ports/SkFontMgr_android.cpp",
+    "src/ports/SkFontMgr_android_factory.cpp",
+    "src/ports/SkFontMgr_android_parser.cpp",
+  }
+
+  -- add common MacOS and iOS
+
+  local common_cocoa = {
+    common_unix,
+    "src/ports/SkFontHost_mac.cpp",
+    "src/utils/mac/SkStream_mac.cpp",
+  }
+
+  -- Opts SSE
+
+  local opts_sse = {
+    "src/opts/SkOpts_ssse3.cpp",
+    "src/opts/SkOpts_sse41.cpp",
+    "src/opts/SkOpts_sse42.cpp",
+    "src/opts/SkOpts_avx.cpp",
+    "src/opts/SkOpts_crc32.cpp",
+    "src/opts/SkOpts_hsw.cpp",
+  }
+
+  -- Opts arm64
+
+  local opts_arm = {
+    "src/opts/SkOpts_crc32.cpp",
+  }
+  
+  flags {
+    "NoPCH",
+  }
+
+  defines {
+    "SK_SUPPORT_GPU=0",
+    "SK_SCALAR_IS_FLOAT",
```

Comment:
This can probably be removed as it is defined globally https://devtopia.esri.com/3rdparty/architecture/blob/runtimecore/premake/static_project.lua#L26. This is defined globally since skia headers use it and we expose skia headers in our classes.

Reply:
Since `skia.lua` brings in `3rdparty_static_project.lua` instead of `static_project.lua`, would we still need to define it here?

Reply:
Ahh you're right